jobs/fxci-taskcluster-export/fxci_etl/loaders/bigquery.py

```diff
@@ -1,23 +1,67 @@
 import base64
 import json
 from abc import ABC, abstractmethod
-from collections import defaultdict
-from dataclasses import InitVar, asdict, dataclass
+from dataclasses import InitVar, asdict, dataclass, fields, is_dataclass
 from datetime import datetime, timezone
 from pprint import pprint
-from typing import Any
+from typing import Any, Type, TypeAlias, Union, get_args, get_origin
 
 import dacite
+from aenum import Enum, NoAlias
+from dacite.config import Config as DaciteConfig
 from google.cloud import storage
 from google.cloud.exceptions import NotFound
-from google.cloud.bigquery import Client, Table
+from google.cloud.bigquery import (
+    Client,
+    SchemaField,
+    Table,
+    TimePartitioning,
+    TimePartitioningType,
+)
 
 from fxci_etl.config import Config
 
 
+class BigQueryTypes(Enum, settings=NoAlias):  # type: ignore
+    DATE: TypeAlias = str
+    FLOAT: TypeAlias = float
+    INTEGER: TypeAlias = int
+    STRING: TypeAlias = str
+    TIMESTAMP: TypeAlias = int
+
+
+def generate_schema(cls):
+    assert is_dataclass(cls)
```

Comment:
I'm not familiar with the codebase so maybe it's common practice: I know Python assertions can easily be turned off by a flag. What's the reason for not raising an error here?

Reply:
Yeah, we could.. Sometimes I use assertions almost like comments or type hints. In this case I didn't know how to do a type hint so figured an `assert` is good and clear enough. If assertions were disabled, this would raise an exception a little later on anyway (in the `fields(cls)` call).